src/d3a/models/area/__init__.py

```diff
@@ -85,17 +88,22 @@ def __init__(self, name: str = None, children: List["Area"] = None,
         if budget_keeper:
             self.budget_keeper.area = self
         self._bc = None
-        self._markets = AreaMarkets(self.log)
-        self.stats = AreaStats(self._markets)
+        self._markets = None
         self.dispatcher = DispatcherFactory(self)()
         self.transfer_fee_pct = transfer_fee_pct
         self.transfer_fee_const = transfer_fee_const
         self.display_type = "Area" if self.strategy is None else self.strategy.__class__.__name__
+        self.redis_ext_conn = RedisExternalConnection(self) \
+            if external_connection_available is True else None
 
     def set_events(self, event_list):
         self.events = Events(event_list, self)
 
     def activate(self, bc=None):
+        # TODO: Validate whether this is right:
```

Comment:
yeah, I moved this from the constructor to the activate method because we only need the markets then. Also: create markets inly for none-leaf areas.

Reply:
No problem for me, everything works good even with this change.

Reply:
@hannesdiedrich on second thought, many unit tests fail just because of that. I will revert this change if it is not necessary with a TODO comment to refactor it in isolation.

Reply:
OK! Agreed!